### ISSUE TITLE
How to debounce using a function with optional parameters?

### ISSUE THREAD

**Issue by felpsio** [localhurst/just_debounce_it#2]
Thanks for the package. I'm always using it and now  I need to debounce a function with optional parameters and I'm not sure what is the best way to do it.

I  have an `updateFilter` function with many optional parameters how should I do it?

Example:
```
Debounce.duration(
    Duration(seconds: 3),
    updateFilter,
    [searchText: newSearchText]
);
```

one option would be:
```
final aux = Function(String searchText) {
   updateFilter(searchText: searchText);
}

Debounce.duration(
    Duration(seconds: 3),
    aux,
    [newSearchText]
);
```

Is there a better option?

**Comment by ghost** [comment]
Right now I think you have the best solution for named parameters. That's a
use-case I had not considered, but it would be nice if the intermediary
function you've created wasn't necessary.

I'll look into ways to support this. Thanks for the open issue! 👍

**Comment by felpsio** [comment]
Great! :D

One option is to add a `Map<String, dynamic>` in the debounce functions. Like:
```
Debounce.duration(
  Duration(seconds: 3),
  callback,
  [parameter1, parameter2, ...],
  {'parameter1': value1, 'parameter2': value2 etc}
)
```

Thanks for the package again :D

**Comment by mhrst** [comment]
_(moved repo to my main github account)_

Just published 3.0.0+1 which includes a new param on `Debounce` functions for providing named parameters. I went with a `Map<Symbol, dynamic>` since dart's `Function` class is already set up to handle parameters that way.

I'd like to allow `String` instead of `Symbol` for ease-of-use but there are some rules to what Strings make valid Function parameter names. Using a Symbols for now is more verbose, but less error-prone.

**Comment by felpsio** [comment]
Awesome! Thank you very much @truthmast :)
